### PR TITLE
Centralize version parsing in mozilla-version 0.4.x

### PR DESCRIPTION
**Pull request by JohanLorenzo**
r? @rail

### CODE REVIEW

**Comment by rail** [comment]
This is awesome! thank you! Let me test this a bit.

File: src/shipit/api/shipit_api/release.py [hunk 1]
```diff
@@ -39,78 +27,73 @@ class ProductCategory(enum.Enum):
...
+    Product.FIREFOX: FirefoxVersion,
+    Product.THUNDERBIRD: ThunderbirdVersion,
+}
 
-def is_final_release(version):
-    return bool(re.match(FINAL_RELEASE_REGEX, version))
 
+def parse_version(product, version):
+    try:
+        product_enum = Product[product.upper()]
```
Comment (rail) [inline src/shipit/api/shipit_api/release.py]: A small issue here.
```suggestion
        product_enum = Product[product.value.upper()]
```

**Comment by JohanLorenzo** [comment]
Thanks for testing it out! I implement a slightly different fix because some other places do rely on `product` to be a string. 

Tell me what you think :)

**Comment by JohanLorenzo** [comment]
Note: this PR will facilitate https://bugzilla.mozilla.org/show_bug.cgi?id=1562629

**Comment by rail** [comment]
Let's merge it after the release, in a couple of days.

**Comment by lundjordan** [comment]
@JohanLorenzo we don't know the root cause but rail pointed out that the partial condition check for determining if `rc` got moved out by one scope: https://github.com/mozilla/release-services/pull/2231/files#diff-ae9b2045812d1ad31281348eea2154a3R71

we saw today's beta become an 'rc' in shipit.

**Comment by rail** [comment]
We should probably just return early from that function, something like

```python
if not gecko_version.is_release or gecko_version.patch_number is not None:
    return False
```

**Pull request merged**